### PR TITLE
[improve] [broker] disable balancing based on DirectMemory.

### PR DESCRIPTION
**Pull request by thetumbled**
### Motivation
As shown in the following figure, the throughput in/out, message in/out, bandwidht in/out usage and CPU usage go up and down periodically as time goes by, while the memory usage and direct memory usage has nothing to do with the actual load.
image
image
image
image

Using memory usage and direct memory usage to score the broker load will result into wrong result.

### Modifications

~~Removing memory usage and direct memory usage for scoring.
As [PR#19559 ](https://github.com/apache/pulsar/pull/19559) has removed memory usage, i will remove direct memory usage for scoring.~~
Change the default value of `loadBalancerDirectMemoryResourceWeight`.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository: https://github.com/thetumbled/pulsar/pull/30

### CODE REVIEW

**Comment by thetumbled** [comment]
Could you help to review this PR? thanks. @heesung-sn @Demogorgon314

**Review by Demogorgon314 — APPROVED** [review]
+1

File: pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java [hunk 1]
```diff
@@ -2414,14 +2414,17 @@ The delayed message index time step(in seconds) in per bucket snapshot segment,
             doc = "Memory Resource Usage Weight. Deprecated: Memory is no longer used as a load balancing item.",
             deprecated = true
     )
-    private double loadBalancerMemoryResourceWeight = 1.0;
+    private double loadBalancerMemoryResourceWeight = 0;
 
+    @Deprecated(since = "3.0.0")
```
Comment (BewareMyPower) [inline pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java:2419]: Should it be 3.2.0?

Comment (thetumbled) [reply pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java]: fixed.

**Comment by heesung-sn** [comment]
Can we keep this direct memory signal but the default `loadBalancerDirectMemoryResourceWeight` to zero? I wonder if direct memory can be useful for some other cases.

**Comment by thetumbled** [comment]
> Can we keep this direct memory signal but the default `loadBalancerDirectMemoryResourceWeight` to zero? I wonder if direct memory can be useful for some other cases.

It is weird that we remove momory signal but retain direct memory signal, right?

**Comment by heesung-sn** [comment]
> > Can we keep this direct memory signal but the default `loadBalancerDirectMemoryResourceWeight` to zero? I wonder if direct memory can be useful for some other cases.
> 
> It is weird that we remove the memory signal but retain direct memory signal, right? IMO, either we retain both or remove both.

IMO, these removals need to be agreed by the community. I don't know if any users rely on these memory and direct memory signals in their production. This seems to be a radical decision.

**Comment by thetumbled** [comment]
> > > Can we keep this direct memory signal but the default `loadBalancerDirectMemoryResourceWeight` to zero? I wonder if direct memory can be useful for some other cases.
> > 
> > 
> > It is weird that we remove the memory signal but retain direct memory signal, right? IMO, either we retain both or remove both.
> 
> IMO, these removals need to be agreed by the community. I don't know if any users rely on these memory and direct memory signals in their production. This seems to be a radical decision.

It does. My original idea is to set the default value of `loadBalancerDirectMemoryResourceWeight`、`loadBalancerMemoryResourceWeight` to be zero. But, i found that PR https://github.com/apache/pulsar/pull/19559 has removed memory signal, so i follow it.
WDYT, @hangc0276 @Demogorgon314

**Comment by hangc0276** [comment]
I think we can change the default value to `0`

**Review by BewareMyPower — APPROVED** [review]

**Review by hangc0276 — APPROVED** [review]

**Review by aloyszhang — APPROVED** [review]

**Review by heesung-sn — APPROVED** [review]

**Comment by lhotari** [comment]
There's a related enhancement proposal in #21973 , please take a look.

**Pull request merged**